packages/terra-table/src/TableSectionHeader.jsx

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import TableUtils from './TableUtils';
+import styles from './TableSectionHeader.module.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Number of columns the section header must span
+   */
+  colSpan: PropTypes.number.isRequired,
+  /**
+   * Whether or not the section is collapsed.
+   */
+  isCollapsed: PropTypes.bool,
+  /**
+   * Whether or not the section can be collapsed.
+   */
+  isCollapsible: PropTypes.bool,
+  /**
+   * Optionally sets the heading level. One of 2`, `3`, `4`, `5`, `6`.
+   */
+  level: PropTypes.oneOf([2, 3, 4, 5, 6]),
+  /**
+   * The associated metaData to be provided in the onSelect callback.
+   */
+  // eslint-disable-next-line react/forbid-prop-types
+  metaData: PropTypes.object,
+  /**
+   * Function callback for when the appropriate click or key action is performed.
+   * Callback contains the javascript evnt and prop metadata, e.g. onSelect(event, metaData)
+   */
+  onSelect: PropTypes.func,
+  /**
+   * Function callback passthrough for the ref of the section li.
+   */
+  refCallback: PropTypes.func,
+  /**
+   * Title text to be placed within the section header.
+   */
+  title: PropTypes.string,
+};
+
+const defaultProps = {
+  isCollapsed: false,
+  isCollapsible: false,
+  level: 2,
+  title: '',
+};
+
+const TableSectionHeader = ({
+  colSpan,
+  isCollapsed,
+  isCollapsible,
+  level,
+  metaData,
+  onBlur,
+  onClick,
+  onKeyDown,
+  onMouseDown,
+  onSelect,
+  refCallback,
+  title,
+  ...customProps
+}) => {
+  const sectionHeaderClassNames = cx([
+    'section-header',
+    { 'is-collapsible': isCollapsible },
+    customProps.className,
+  ]);
+
+  const attrSpread = {};
+  const Element = `h${level}`;
+  let titleElement = <Element className={cx('title')}>{title}</Element>;
```

Comment:
if `title` isn't required, we could conditionally render this

Reply:
Good catch, I'll update.

Reply:
Update to be required.